openedx/core/lib/inline_analytics_utils.py

```diff
@@ -70,18 +84,59 @@ def get_responses_data(block):
             if part_id and part_id in valid_responses and node.tag in valid_group_nodes:
                 # This is a valid question according to the list of valid responses and we have the group node
 
                 if valid_responses[part_id].response_type == 'other':
                     # Response type is not supported by the analytics api
-                    responses_data.append(AnalyticsContextResponse(part_id, None, None, "The analytics cannot be displayed for this type of question.", None))
+                    message = _('The analytics cannot be displayed for this type of question.')
+                    responses_data.append(AnalyticsContextResponse(part_id,
+                                                                   None,
+                                                                   None,
+                                                                   message,
+                                                                   None))
                 elif rerandomize:
                     # Response, actually the problem, has rerandomize != 'never'
-                    responses_data.append(AnalyticsContextResponse(part_id, valid_responses[part_id].correct_response, valid_responses[part_id].response_type, "The analytics cannot be displayed for this question as it uses randomization.", None))
+                    message = _('The analytics cannot be displayed for this question as it uses randomization.')
+                    responses_data.append(AnalyticsContextResponse(part_id,
+                                                                   valid_responses[part_id].correct_response,
+                                                                   valid_responses[part_id].response_type,
+                                                                   message,
+                                                                   None))
                 else:
                     # Response is supported by the analytics api and rerandomize == 'never'
-                    responses_data.append(AnalyticsContextResponse(part_id, valid_responses[part_id].correct_response, valid_responses[part_id].response_type, None, cgi.escape(json.dumps(valid_responses[part_id].choice_name_list), quote=True)))
+                    return_data = cgi.escape(json.dumps(valid_responses[part_id].choice_name_list), quote=True)
+                    responses_data.append(AnalyticsContextResponse(part_id,
+                                                                   valid_responses[part_id].correct_response,
+                                                                   valid_responses[part_id].response_type,
+                                                                   None,
+                                                                   return_data))
     return responses_data
+
+
+def add_inline_analytics(_user, block, _view, frag, _context):  # pylint: disable=unused-argument
```

Comment:
I may be missing something, but it seems unnecessary to pass in _user, _view, and _context.  Does this function use them for anything?